src/test/java/org/broadinstitute/hellbender/tools/coveragemodel/germline/CoverageModellerGermlineSparkToggleIntegrationTest.java

```diff
@@ -0,0 +1,398 @@
+
+        Assert.assertTrue(String.format("The homozygous deletion calling concordance %f is lower than expected threshold %f",
+                averageAbsConcordance[0], MIN_PASS_HOM_DEL_CONCORDANCE), averageAbsConcordance[0] > MIN_PASS_HOM_DEL_CONCORDANCE);
+
+        Assert.assertTrue(String.format("The average absolute copy number calling concordance %f is lower than expected threshold %f",
+                averageOverallAbsConcordance, MIN_PASS_ABS_CONCORDANCE), averageOverallAbsConcordance > MIN_PASS_ABS_CONCORDANCE);
+    }
+
+    private SampleCopyNumberConcordanceSummary calculateSampleCopyNumberConcordance(final ReadCountCollection truth,
+                                                                                                                             final ReadCountCollection inferred,
+                                                                                                                             final int sampleIndex,
+                                                                                                                             final String sampleSexGenotype) {
+        final List<Target> targets = truth.targets();
+        int targetIndex = 0;
+
+        final int[] truthCopyNumberCallsPerSample = Arrays.stream(truth.getColumn(sampleIndex))
+                .mapToInt(d -> (int)d).toArray();
+        final int[] inferredCopyNumberCallsPerSample = Arrays.stream(inferred.getColumn(sampleIndex))
+                .mapToInt(d -> (int)d).toArray();
+
+        final int[] correctCopyNumberCallCounts = new int[MAX_COPY_NUMBER + 1];
+        final int[] truthCopyNumberCallCounts = new int[MAX_COPY_NUMBER + 1];
+        int truthRefCallCounts = 0, truthAltCallCounts = 0;
+        int correctRefCallCounts = 0, correctAltCallCounts = 0;
+        for (final Target target : targets) {
+            final int refCopyNumber = GERMLINE_PLOIDY_ANNOTATIONS.getTargetGermlinePloidyByGenotype(target,
+                    sampleSexGenotype);
+            final int truthCopyNumber = truthCopyNumberCallsPerSample[targetIndex];
+            final int inferredCopyNumber = inferredCopyNumberCallsPerSample[targetIndex];
+
+            /* ref/alt-blind concordance */
+            truthCopyNumberCallCounts[truthCopyNumber]++;
+            if (inferredCopyNumber == truthCopyNumber) {
+                correctCopyNumberCallCounts[truthCopyNumber]++;
+            }
+
+            /* ref/alt-resolved concordance */
+            if (truthCopyNumber == refCopyNumber) { /* ref truth */
+                truthRefCallCounts++;
+                if (inferredCopyNumber == truthCopyNumber) {
+                    correctRefCallCounts++;
+                }
+            } else { /* alt truth */
+                truthAltCallCounts++;
+                if (inferredCopyNumber == truthCopyNumber) {
+                    correctAltCallCounts++;
+                }
+            }
+            targetIndex++;
```

Comment:
This code will be more maintainable as a `for` loop over `targetIndex`.  For clarity (at a small cost in speed), I would also consider breaking it up into a few `. . ..stream(). . .count()` statements if possible.